yara-x/src/modules/string.rs

```diff
@@ -0,0 +1,86 @@
+use crate::modules::prelude::*;
+use crate::modules::protos::string::*;
+
+#[module_main]
+fn main(_ctx: &ScanContext) -> String {
+    // Nothing to do, but we have to return our protobuf
+    let string_proto = String::new();
+    string_proto
+}
+
+#[module_export(name = "to_int")]
+fn to_int(ctx: &ScanContext, string: RuntimeString) -> Option<i64> {
+    let string = string.as_bstr(ctx);
+
+    match string.to_str() {
```

Comment:
Here you can take advantage of `Result::ok()` and the `?` operator, for example:

```rust
#[module_export(name = "to_int")]
fn to_int(ctx: &ScanContext, string: RuntimeString) -> Option<i64> {
    let string = string.as_bstr(ctx).to_str().ok()?;
    string.parse::<i64>().ok()
}
```

The `.ok()` method takes a `Result<T, E>` and returns an `Option<T>` if no error occurred, or `None` if otherwise. The `?` operator propagates the `None` to the caller, and returns `T`.

So, the first line returns the `&str` you need in a more compact way, while the second line parses the `i64` and returns its, or returns `None` if an error occurred.

I also think it's worth adding a `.to_str()` method to `RuntimeString`, it would be more legible:

```rust
fn to_int(ctx: &ScanContext, string: RuntimeString) -> Option<i64> {
    let string = string.to_str().ok()?;
    string.parse::<i64>().ok()
}
```

Reply:
I've added a `to_str()` method to `RuntimeString`. This is possible now:

```
    let string = string.to_str(ctx).ok()?;
    string.parse::<i64>().ok()
```

Reply:
Thanks for the feedback. I will get it fixed up.

Reply:
Fixed. Thanks